week1/plot_trips.R

```diff
@@ -47,16 +100,83 @@ trips_with_weather <- inner_join(trips, weather, by="ymd")
 # you'll need to decide what constitutes "substantial precipitation" and create a new T/F column to indicate this
+trips_weather$subs_precip <- trips_weather$prcp >= 0.1
+  
+ggplot(trips_weather, aes(x = tmin, y = count, label = ymd)) +
+  geom_point() +
+  geom_text(position = "identity", size = 2, vjust = 1.5) +
+  facet_wrap(~ subs_precip) +
+  xlab("Minimum Temperature") +
+  ylab("Number of Trips") 
 
 # add a smoothed fit on top of the previous plot, using geom_smooth
+ggplot(trips_weather, aes(x = tmin, y = count, label = ymd)) +
+  geom_point() +
+  geom_text(position = "identity", size = 2, vjust = 1.5) +
+  geom_smooth(method = "lm") +
+  scale_y_log10(label = comma) +
+  facet_wrap(~ subs_precip) +
+  xlab("Minimum Temperature") +
+  ylab("Number of Trips")
 
 # compute the average number of trips and standard deviation in number of trips by hour of the day
 # hint: use the hour() function from the lubridate package
+trips_with_weather$hour <- hour(trips_with_weather$starttime)
+
+summ_by_hour <- trips_with_weather %>%
+  group_by(ymd,hour) %>% 
+  summarize(count = n())
 
+hours_summary <- group_by(summ_by_hour,hour) %>% 
+  summarize(average_num_trips = mean(count),
+            sd_num_trips = sd(count))
+head(hours_summary)
+  
 # plot the above
+ggplot(hours_summary) +
+ # geom_point(aes(x = hour,  y = average_num_trips, color = "average num. of trips")) + 
+  #geom_point(aes(x = hour, y = sd_num_trips, color = "standard deviation")) + 
+  geom_line(aes(x = hour, y = average_num_trips)) +
+  geom_ribbon(aes(x = hour, ymin = average_num_trips - sd_num_trips, ymax = average_num_trips + sd_num_trips), alpha = 0.2) + 
+  xlab("Hour of Day") + 
+  ylab("Num of Trips") + 
+ # scale_color_manual("",
+  #                   breaks = c("average num. of trips", "standard deviation"),
+   #                  values = c("red", "blue"))
 
 # repeat this, but now split the results by day of the week (Monday, Tuesday, ...) or weekday vs. weekend days
 # hint: use the wday() function from the lubridate package
+trips_with_weather$day_of_week <- wday(trips_with_weather$ymd)
+
+summ_by_day <- trips_with_weather %>%
+  group_by(ymd,day_of_week) %>%
+  summarize(count = n())
+
+day_summary <- summarize(group_by(summ_by_day,day_of_week),
+            average_num_trips = mean(count),
+            sd_num_trips = sd(count))
+
+ggplot(day_summary) + 
+  geom_line(aes(x = day_of_week, y = average_num_trips)) + 
+  geom_ribbon(aes(x = day_of_week, ymin = average_num_trips - sd_num_trips, ymax = average_num_trips + sd_num_trips), alpha = 0.2) +
+ # geom_point(aes(x = day_of_week, y = average_num_trips, color = "average num. of trips")) + 
+  #geom_point(aes(x = day_of_week, y = sd_num_trips, color = "standard deviation")) + 
+  xlab("Day of the Week") + 
+  ylab("Num of Trips") 
+ # scale_color_manual("", 
+  #                   breaks = c("average num. of trips", "standard deviation"),
+   #                  values = c("red", "blue"))
```

Comment:
For this last problem we actually wanted it split the other way, so that you get a separate time of day plot for each day of the week. That being said, you did the math right it's just that we wanted facet_wrap() at the end